py/kubeflow/testing/new_test_util.py

```diff
@@ -0,0 +1,155 @@
+"""
+This module contains misc utils used for Kubeflow testing
+infrastructure. They include creating test cases/suites,
+managing logs of test cases and initializing tests
+
+This is intended to replace test_util.py
+"""
+import argparse
+import logging
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+
+import junit_xml
+
+
+class TestCase(junit_xml.TestCase):
+  """A single test case."""
+
+  def __init__(self, test_func, **kwargs):
+    self.class_name = "Kubeflow"
+    self.start_time = time.time()
+    self.test_func = test_func
+    super(TestCase, self).__init__(**kwargs)
+
+  def add_failure_info(self, message=None, output=None, failure_type=None):
```

Comment:
please add doc. what's this function for and what is each arg.

Reply:
Done